### PR TITLE
Adds ptiff derivative support

### PR DESCRIPTION
**Pull request by mlooney**

### CODE REVIEW

File: lib/tasks/derivatives.rake [hunk 1]
```diff
@@ -14,12 +14,16 @@ namespace :derivatives do
...
+    FileSetAttachedEventJob.new.perform(FileSet.find(ENV.fetch("fsid")), User.find(ENV.fetch("uid")))
+  end
+
   def regenerate_derivatives(work)
     work.file_sets.each do |fs|
-      puts " processing FileSet #{fs.id}"
       asset_path = fs.original_file.uri.to_s
       asset_path = asset_path[asset_path.index(fs.id.to_s)..-1]
-      CreateDerivativesJob.perform_later(fs, asset_path)
+      CreateDerivativesJob.new.perform(fs, asset_path)
```
Comment (maxkadel) [inline lib/tasks/derivatives.rake:26]: This means that it won't run as a background job, which also means it won't re-run if it fails, is that what we want?

**Review by maxkadel — DISMISSED** [review]
I think even for the rake task we want to run it as a background job. 

I'd need to be walked through the `app/extensions/derivatives`, right now I can't follow it.

File: lib/tasks/derivatives.rake (hunk 1 above)
Comment (mark-dce) [reply lib/tasks/derivatives.rake:26]: I'm not sure jobs retrying automatically is always the thing we want to happen.  I think if we're running this in the context of rake task that means
1) we're running it from the command line and we have other methods of backgrounding the process
2) we might want more immediate feedback about failures
3) we probably need to manually intervene for the jobs to ever run successfully anyway

File: spec/extensions/derivatives.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+require './app/extensions/derivatives.rb'
+RSpec.describe TenejoExtensions::PtiffDerivative, :clean do
+  let(:fsds) {
+    fs = instance_double("fileset")
+    allow(fs).to receive(:id).and_return("abc")
+    allow(fs).to receive(:mime_type).and_return("image/jpeg")
+    Hyrax::FileSetDerivativesService.new(fs)
+  }
+  it "creates 2 outputs" do
+    r = fsds.send(:create_image_derivatives, "./spec/fixtures/images/small.jpg")
```
Comment (mark-dce) [inline spec/extensions/derivatives.rb]: What is an `r`?  Can we give this a more descriptive name like `derivative`

File: spec/extensions/derivatives.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    expect(r.size).to eq(2)
+    expect(r.first[:url]).to match(/\/.*-ptiff.tif/)
+    expect(r.first[:label]).to match(/ptiff/)
+    expect(r.first[:format]).to match(/ptif/)
+  end
+
+  it "actually creates files" do
+    r = fsds.send(:create_image_derivatives, "./spec/fixtures/images/small.jpg")
+    u = r[0][:url].gsub("file://.", "")
```
Comment (mark-dce) [inline spec/extensions/derivatives.rb]: Could you give `r` and `u` more descriptive names here too?

**Review by mark-dce — CHANGES_REQUESTED** [review]
In general, I'd feel more comfortable maintaining this code in the future if some of the variable names were more descriptive.  It would especially help me read through and make sense of the tests more quickly.

File: lib/tasks/derivatives.rake (hunk 1 above)
Comment (mlooney) [reply lib/tasks/derivatives.rake:26]: to be fair, I didn't mean to commit this.  but also, I agree with mark.

**Review by mark-dce — APPROVED** [review]

**Pull request merged**